examples/sol-contract/src/processor.rs

```diff
@@ -0,0 +1,110 @@
+//! Program instruction processor
+//! Only the program admin can issue the Init instruction.
+//! And anyone can check the loan with the Loan2Value instruction.
+
+use solana_program::account_info::{
+    next_account_info,
+    AccountInfo,
+};
+use solana_program::entrypoint::ProgramResult;
+use solana_program::msg;
+use solana_program::program_error::ProgramError;
+use solana_program::program_pack::{
+    IsInitialized,
+    Pack,
+};
+use solana_program::pubkey::Pubkey;
+
+use borsh::BorshDeserialize;
+use pyth_sdk_solana::load_price_feed_from_account_info;
+
+use crate::instruction::ExampleInstructions;
+use crate::state::AdminConfig;
+
+pub fn process_instruction(
+    _program_id: &Pubkey,
+    _accounts: &[AccountInfo],
+    input: &[u8],
+) -> ProgramResult {
+    let account_iter = &mut _accounts.iter();
+    let signer = next_account_info(account_iter)?;
+    let data_account = next_account_info(account_iter)?;
```

Comment:
I suggest renaming this to `admin_config_account` or something similar.

Reply:
Also it's probably better that the admin config account be a PDA of the program.